code-matlab/shared/util/restrict3.m

```diff
@@ -0,0 +1,93 @@
+function in = restrict3(in,varargin)
+% function out = restrict(in,varargin)
+%
+% restricts times in data object to specific intervals
+%
+% if a data time is equal to an interval edge, this data point is INCLUDED
+%
+% usage:
+% out = restrict(in,iv)
+% out = restrict(in,tstart,tend) % tstart and tend can be vectors
+%
+% works on ts, tsd, iv data
+%
+% note that for iv data, intervals must fit within restrict times (i.e.
+% will not get cut)
+%
+% MvdM 2014-07-20 initial version
+% youkitan 2016-11-27 edit: type checking update
```

Comment:
a brief comment explaining the difference between restrict, restrict2 and 3 would be useful

Reply:
Restrict was really slow so I had implemented faster versions of it for different datatypes and had forgotten about it. To be honest, restrict2 and restrict3 should be merged since they have different algorithms for different datatypes (restrict2 for tsd, restrict3 for ts). Plus, they aren't tested for all datatyes (ts,iv and tsd) so it's probably better to stick with the original restrict. I will combine the two functions and test it properly for another merge.